concent_api/core/tasks.py

```diff
@@ -229,6 +278,21 @@ def verification_result(
                 subtask=subtask,
             )
 
+        task_to_compute = deserialize_message(subtask.task_to_compute.data.tobytes())
+
+        delete_deposit_claim(
+            subtask_id=subtask_id,
+            concent_use_case=ConcentUseCase.ADDITIONAL_VERIFICATION,
+            ethereum_address=task_to_compute.requestor_ethereum_address,
+            public_key=task_to_compute.requestor_public_key,
+        )
+        delete_deposit_claim(
```

Comment:
I am not sure if I understand correctly. If there's `VERIFICATION_MISMATCH`, why provider's claim is deleted? I thought that provider should pay for verification anyways.

Reply:
Same as above, plus there was no payment here before.

Reply:
@dybi is right. If there's a mismatch:
- claim against requestor's account should be deleted (provider did not deliver the correct result; no need to pay)
- claim against provider's account should be finalized (Concent did deliver it's service by issuing a verdict and needs to be paid).

In general the claim against the provider (i.e. the verification fee) should never be discarded if verification ends with a verdict (i.e. when Concent performs its service successfully). Currently we give a verdict even if verification fails due to our own fault or times out so this means that provider always pays.

I remember that we talked about it with Łukasz at some point and he said that there should be a separate case for Concent's failure. I'll need to check if that's been changed in the spec or not. Nevertheless, Concent's failure or timeout should be a rare case so it does not matter all that much. The current approach is fine for the time being.